### PR TITLE
IAM: Provider resource overhaul

### PR DESCRIPTION
**Pull request by artem-lifshits**
## Summary of the Pull Request
Implement new resource ``opentelekomcloud_identity_provider`` to manage identity providers with SAML/OIDC support

## PR Checklist

* [x] Refers to: #2177
* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.
* [x] Release notes added.

## Acceptance Steps Performed

```
=== RUN   TestAccIdentityProviderBasic
--- PASS: TestAccIdentityProviderBasic (50.26s)
=== RUN   TestAccIdentityProviderOIDC
--- PASS: TestAccIdentityProviderOIDC (51.19s)
=== RUN   TestAccIdentityOIDCProvider_import
--- PASS: TestAccIdentityOIDCProvider_import (33.64s)
=== RUN   TestAccIdentitySamlProvider_import
--- PASS: TestAccIdentitySamlProvider_import (35.68s)
PASS

Process finished with the exit code 0

```

### CODE REVIEW

**Review by vladimirvshivkov — DISMISSED** [review]
looks fine

**Review by Polina-Gubina — DISMISSED** [review]

**Review by vladimirvshivkov — DISMISSED** [review]

**Review by Polina-Gubina — DISMISSED** [review]

**Review by vladimirvshivkov — DISMISSED** [review]

File: opentelekomcloud/services/iam/resource_opentelekomcloud_identity_provider.go [hunk 1]
```diff
@@ -0,0 +1,578 @@
...
+			"metadata": {
+				Type:      schema.TypeString,
+				Optional:  true,
+				StateFunc: common.GetHashOrEmpty,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"access_config": {
```
Comment (anton-sidelnikov) [inline opentelekomcloud/services/iam/resource_opentelekomcloud_identity_provider.go:83]: conflict with metadata

File: opentelekomcloud/services/iam/resource_opentelekomcloud_identity_provider.go [hunk 2]
```diff
@@ -0,0 +1,578 @@
...
+			mappings.Delete(client, conversionRuleID).ExtractErr(),
+		)
+		log.Printf("[ERROR] Error creating protocol, and the mapping that has been created. Error: %s", mErr)
+		return fmt.Errorf("error creating identity provider protocol: %s", mErr.Error())
+	}
+	return nil
+}
+
+func getDefaultConversionOpts() *mappings.CreateOpts {
+	localRules := []mappings.LocalRuleOpts{
```
Comment (anton-sidelnikov) [inline opentelekomcloud/services/iam/resource_opentelekomcloud_identity_provider.go:353]: try to make rules editable

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

File: opentelekomcloud/services/iam/resource_opentelekomcloud_identity_provider.go (hunk 1 above)
Comment (artem-lifshits) [reply opentelekomcloud/services/iam/resource_opentelekomcloud_identity_provider.go:83]: done

File: opentelekomcloud/services/iam/resource_opentelekomcloud_identity_provider.go (hunk 2 above)
Comment (artem-lifshits) [reply opentelekomcloud/services/iam/resource_opentelekomcloud_identity_provider.go:353]: done

**Review by Aloento — APPROVED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Pull request merged**